### ISSUE TITLE
Are there any performance issues with new version 0.1.1?

### ISSUE THREAD

**Issue by etzimopoulos** [dexplo/jupyter_to_medium#9]
Hi Ted,

I upgraded this morning to latest version 0.1.1. and I'm getting two issues:
1) slow running of the publishing - takes significantly and I can see it from the logs
2) random timeouts and failure to publish. e.g. 

```
loading image to medium
loading image to medium
SSLError: HTTPSConnectionPool(host='api.medium.com', port=443): Max retries exceeded with url: /v1/images (Caused by SSLError(SSLError("bad handshake: SysCallError(-1, 'Unexpected EOF')")))

Traceback (most recent call last):
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/urllib3/contrib/pyopenssl.py", line 485, in wrap_socket
    cnx.do_handshake()
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/OpenSSL/SSL.py", line 1934, in do_handshake
    self._raise_ssl_error(self._ssl, result)
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/OpenSSL/SSL.py", line 1664, in _raise_ssl_error
    raise SysCallError(-1, "Unexpected EOF")
OpenSSL.SSL.SysCallError: (-1, 'Unexpected EOF')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/urllib3/connectionpool.py", line 672, in urlopen
    chunked=chunked,
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/urllib3/connectionpool.py", line 376, in _make_request
    self._validate_conn(conn)
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/urllib3/connectionpool.py", line 994, in _validate_conn
    conn.connect()
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/urllib3/connection.py", line 360, in connect
    ssl_context=context,
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/urllib3/util/ssl_.py", line 370, in ssl_wrap_socket
    return context.wrap_socket(sock, server_hostname=server_hostname)
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/urllib3/contrib/pyopenssl.py", line 491, in wrap_socket
    raise ssl.SSLError("bad handshake: %r" % e)
ssl.SSLError: ("bad handshake: SysCallError(-1, 'Unexpected EOF')",)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/urllib3/connectionpool.py", line 720, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/urllib3/util/retry.py", line 436, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='api.medium.com', port=443): Max retries exceeded with url: /v1/images (Caused by SSLError(SSLError("bad handshake: SysCallError(-1, 'Unexpected EOF')")))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/jupyter_to_medium/_bundler.py", line 37, in upload
    data = publish(**kwargs)
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/jupyter_to_medium/_publish_to_medium.py", line 300, in publish
    p.main()
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/jupyter_to_medium/_publish_to_medium.py", line 220, in main
    self.load_images_to_medium()
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/jupyter_to_medium/_publish_to_medium.py", line 142, in load_images_to_medium
    r = requests.post(self.IMAGE_URL, headers=self.headers, files=file_payload)
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/home/etzimopoulos/anaconda3/lib/python3.7/site-packages/requests/adapters.py", line 514, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='api.medium.com', port=443): Max retries exceeded with url: /v1/images (Caused by SSLError(SSLError("bad handshake: SysCallError(-1, 'Unexpected EOF')")))
```
3. Also I've noticed there's more logging about the images being stored from Medium: could that be a reason for slowing it down?

```
Image Storage Information from Medium
-------------------------------------

[
    {
        "data": {
            "url": "https://cdn-images-1.medium.com/proxy/1*-DWXFCNixl3ybo7reWub6w.png",
            "md5": "-DWXFCNixl3ybo7reWub6w"
        }
    },
    {
        "data": {
            "url": "https://cdn-images-1.medium.com/proxy/1*ec8dAfKWNLP499Xs9IR6Tw.png",
            "md5": "ec8dAfKWNLP499Xs9IR6Tw"
        }
    },
    {
        "data": {
            "url": "https://cdn-images-1.medium.com/proxy/1*jEwx8tuASezwkvRBiC3Vbw.png",
            "md5": "jEwx8tuASezwkvRBiC3Vbw"
        }
    },
    {
        "data": {
            "url": "https://cdn-images-1.medium.com/proxy/1*ivYBv91Ajk_mNRR-3eu2vA.png",
            "md5": "ivYBv91Ajk_mNRR-3eu2vA"
        }
    },
    {
        "data": {
            "url": "https://cdn-images-1.medium.com/proxy/1*vxgShHsouSfqfZ349TFqHQ.png",
            "md5": "vxgShHsouSfqfZ349TFqHQ"
        }
    },
    {
        "data": {
            "url": "https://cdn-images-1.medium.com/proxy/1*29RsvtUudavYaS7-pk_vVg.png",
            "md5": "29RsvtUudavYaS7-pk_vVg"
        }
    },
    {
        "data": {
            "url": "https://cdn-images-1.medium.com/proxy/1*JfR2dImWdMXnRXp5akBDtg.png",
            "md5": "JfR2dImWdMXnRXp5akBDtg"
        }
    }
]

```

4. I also switched back to 0.1.0 that worked for me yesterday as initially I wasn't able to get some of the files that worked yesterday to work on 0.1.1. Eventually they're working but sometimes fail and sometimes work very slowly.

5. Finally I tried to run a full data science notebook to see how it works (with more than 20+ images) and I only got one error related to a JPG image saying "We don't know what kind of image this is" or something like that. Once I saved it as PNG it worked. Is there an issue with the JPG format?

**Comment by tdpetrou** [comment]
Definitely upgrade to 0.1.1. It will actually be slightly faster as I changed the handling of images. But, the main bottleneck is the transmission of images to Medium and performance is completely up to their servers as to how they are processing the images.

I've gotten that timeout error from Medium once before. Just wait a couple minutes and then try again.

I added logging of the images from Medium to the output. It's just info printed to the screen and takes no time at all. Previously, it was saved to a local file, but I decided to just print it to the screen now.

In 0.1, JPG images failed. I fixed this bug in 0.1.1.
